caom2pipe/manage_composable.py

```diff
@@ -2924,3 +2925,146 @@ def find_missing(compare_this, to_this):
     :return: list of elements from to_this not in compare_this
     """
     return list(set(compare_this).difference(to_this))
+
+
+class ValueRepairCache(Cache):
```

Comment:
Why the dependency on `Cache`? Can this be a generic `ObservationCurator`, which takes and observation and corrects some values based on a config file?

In the past, we've had (potential?) issues with overriding the metadata, in the sense that users were querying for a keyword, download the file just to discover that the original keyword was different. It's probably not the case with `CAOM2` which is a model derived from FITS metadata but not expected to map directly. Just a comment.

Reply:
The dependence on `Cache` is there because the configuration file name, location identification, and reading/writing, are all handled by that class. 

It also inherits from `Cache` because that's how the *2caom2 works.

Are there better ways to handle knowing what the name of the configuration  file is, where to find it, what it's format should be (yaml is consistent with *2caom2 as well). I picked a way to do those things a few years ago, but I made a lot of bad coding decisions a few years ago too, so changing any and all of those things is within the scope of this discussion. I've even thought this could be worked into a Visitor that would be able to change a lot of things without writing any code, but how would that need to be organized to make a cache file work?

I did not know that users have been surprised by data retrieved from CADC. That's interesting, and I'm not sure what to do about it.

You may know that most of the test cases that work with this code are directly related to changes I've had to made to records, and I've gotten tired of writing specific code for it:
- change release date to later for CFIS
- make Observation.type values all upper case
- change URIs collection/scheme
 What others have you had to do?

Reply:
It makes sense. I was just confused about the name containing the word `Cache`. Does `Cache` have a special connotation in *2caom2?